R/data_modify.R

```diff
@@ -145,6 +149,11 @@ data_modify.default <- function(data, ...) {
 data_modify.data.frame <- function(data, ..., .if = NULL, .at = NULL, .modify = NULL) {
   dots <- eval(substitute(alist(...)))
 
+  # error for data frames with no rows...
+  if (nrow(data) == 0) {
+    insight::format_error("`data` is an empty data frame. `data_modify()` only works for data frames with at least one row.") # nolint
+  }
```

Comment:
Should we error here? I can imagine that this can be used in functions where the input data is not always of the right format and can have 0 rows. I think a better output would be to return a 0-row dataframe with the new columns, like:

``` r
dplyr::mutate(data.frame(x = character(0)), y = 1)
#> [1] x y
#> <0 rows> (or 0-length row.names)
```

My comment was more to ensure that the specific `n()` works as expected when there are 0 rows.

Reply:
Would you prefer just returning an empty data frame then, or give a message/warning and return an empty data frame?

Reply:
Actually, the behavior of `base` is to error, while `dplyr` and `data.table` silently return a 0-row dataframe. I think we can follow the base behavior and error here, so forget this comment.

``` r
library(data.table)
library(dplyr, warn.conflicts = FALSE)

df <- data.frame(x = character(0))
dt <- data.table(x = character(0))
tib <- tibble(x = character(0))

df[["y"]] <- sum(1:3)
#> Error in `[[<-.data.frame`(`*tmp*`, "y", value = 6L): replacement has 1 row, data has 0
dt[, y := sum(1:3)][]
#> Empty data.table (0 rows and 2 cols): x,y
tib |> mutate(y = sum(1:3))
#> # A tibble: 0 × 2
#> # ℹ 2 variables: x <chr>, y <int>
```